### PR TITLE
[REVIEW] Fix strings::pad when using pad::both with odd width

### LINKED ISSUE #4354: [FEA] strings libcudf pad api `side=both` setting needs to align with pandas behavior
**Describe the bug**
Our `pad` api seems to be adding padding differently when `side=both` is passed. See below the comparison. 

**Steps/Code to reproduce bug**
When there is space to pad and **side** selection is **both** pandas seem to first do it from left, whereas we seem to perform from right.
```python
>>> import cudf
>>> s = cudf.Series(["hello","goodbye","well"])
>>> s = cudf.Series(['A,,B', '1,,5', '3,00,0'])
>>> s.str.pad(5, side="both",fillchar='.')
0     A,,B.
1     1,,5.
2    3,00,0
dtype: object
>>> s.to_pandas().str.pad(5, side="both",fillchar='.')
0     .A,,B
1     .1,,5
2    3,00,0
dtype: object

```
**Expected behavior**
We should be aligning with pandas behavior.

**Environment overview (please complete the following information)**
 - Environment location: Docker
 - Method of cuDF install: from source

**Environment details**
Output of the `cudf/print_env.sh` script here: [env.txt](https://github.com/rapidsai/cudf/files/4298969/env.txt)

**Additional context**
This can be tested using PR: #4311

### PR DESCRIPTION
**Pull request by davidwendt**
Closes #4354 
Passing pad widths in Pandas `center()` results in different behavior if the width parameter is an odd vs even value. If the width is odd and the resulting pad is also odd (string width is even), then the extra padding character goes on the _left_ of the resulting string. If the width is even and the resulting pad is odd (string width is odd), then the extra padding character goes on the _right_ of the resulting string.

```
ps = pd.Series(["fox","foxx"])
ps.str.center(6,"+")
0    +fox++
1    +foxx+

ps.str.center(7,"+")
0    ++fox++
1    ++foxx+
```
Using width=6 and odd-lengthed string 'fox' the extra padding char '+' goes on the right.
Using width=7 and even-lengthed string 'foxx' the extra padding char '+" goes on the left.

More examples are in the #4354 description and comments.
Also added more tests that include more pad::BOTH cases.

### CODE REVIEW

**Comment by davidwendt** [comment]
rerun tests

File: cpp/src/strings/padding.cu [hunk 1]
```diff
@@ -124,8 +124,8 @@ std::unique_ptr<column> pad( strings_column_view const& strings,
                 string_view d_str = d_strings.element<string_view>(idx);
                 char* ptr = d_chars + d_offsets[idx];
                 int32_t pad = static_cast<int32_t>(width - d_str.length());
-                auto right_pad = pad/2;
-                auto left_pad = pad - right_pad;
+                auto right_pad = (width & 1) ? pad/2 : (pad-pad/2); // odd width = right-justify
```
Comment (vuule) [inline cpp/src/strings/padding.cu]:
```suggestion
                auto right_pad = (width & 1) ? pad/2 : (pad - pad/2); // odd width = right-justify
```

File: cpp/tests/strings/pad_tests.cu [hunk 2]
```diff
@@ -59,13 +59,31 @@ TEST_F(StringsPadTest, Padding)
...
         cudf::test::strings_column_wrapper expected( h_expected.begin(), h_expected.end(),
              thrust::make_transform_iterator( h_expected.begin(), [] (auto str) { return str!=nullptr; }));
         cudf::test::expect_columns_equal(*results,expected);
     }
 }
 
+TEST_F(StringsPadTest, PaddingBoth)
+{
+    cudf::test::strings_column_wrapper strings( {"koala", "foxx", "fox", "chameleon"} );
+    std::string phil = "+";
```
Comment (harrism) [inline cpp/tests/strings/pad_tests.cu]: A string named Phil?

**Comment by davidwendt** [comment]
rerun tests

**Pull request merged**